### DISCUSSION TITLE
How Do I Resolve This Prisma relationship schema error

### DISCUSSION THREAD

**Discussion by aaronksaunders** [prisma/prisma#13341] [Q&A]
```
model Match {
  id          String   @id @unique @default(cuid())
  name        String
  description String
  player1     User     @relation( fields: [player1Id], references: [id])
  player1Id   String
  player2     User     @relation( fields: [player2Id], references: [id])
  player2Id  String
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  leagueId    String
}


model User {
  id          String        @id @unique @default(uuid())
  email       String        @unique
  password    String
  createdAt   DateTime      @default(now())
  updatedAt   DateTime      @updatedAt
  matches     Match[]       
}
```

Getting following error

```
Error: Schema parsing
error: Error validating model "Match": Ambiguous relation detected. The fields `player1` and `player2` in model `Match` both refer to `User`. Please provide different relation names for them by adding `@relation(<name>).
  -->  schema.prisma:85
   | 
84 |   description String
85 |   player1     User     @relation(fields: [player1Id], references: [id])
86 |   player1Id   String
   | 
```

want to keep track of all of the matches the user plays, each match has two players (Users)

Thanks in advance

**Comment by austincrim** [comment]
Hey @aaronksaunders 👋,

Since `player1` and `player2` both reference the `Player` model, we need to distinguish the between the two relations by giving them distinct **names**. 

The specific names are up to you, but it would look something like this:
```prisma
model Match {
  id          String   @id @unique @default(cuid())
  name        String
  description String
  player1     User     @relation(fields: [player1Id], references: [id], name: "player1")
  player1Id   String
  player2     User     @relation(fields: [player2Id], references: [id], name: "player2")
  player2Id   String
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  leagueId    String
}

model User {
  id             String   @id @unique @default(uuid())
  email          String   @unique
  password       String
  createdAt      DateTime @default(now())
  updatedAt      DateTime @updatedAt
  player1Matches Match[]  @relation("player1")
  player2Matches Match[]  @relation("player2")
}
```

Notice we also had to update the `User` model to distinguish which `Matches` the `User` has been apart of. Let me know if I can clarify anything!

**Reply by aaronksaunders** [reply]
First, thanks for responding... I guess I didn't explain myself correctly
- A User has a list of matches (Matches) that they participated in.
- A match has two players ( Users )

It isn't intuitive that I would need to aggregate `player1Matches` and `player2Matches` to get a list of all of the matches for a specific user

**Reply by aaronksaunders** [reply]
@austincrim after looking through more topics in discussions, it looks like this is the only way to solve this in Prisma

**Reply by austincrim** [reply]
Thanks for the clarification, Aaron. Another solution would be to make it a full many-to-many relation:
```prisma
model Match {
  id          String   @id @unique @default(cuid())
  name        String
  description String
  players     User[]
  createdAt   DateTime @default(now())
  updatedAt   DateTime @updatedAt
  leagueId    String
}

model User {
  id        String   @id @unique @default(uuid())
  email     String   @unique
  password  String
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  matches   Match[]
}
```

I'm not sure how that impacts your business logic, but maybe something to consider.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
